sync/class.jetpack-sync-actions.php

```diff
@@ -53,11 +64,52 @@ static function init() {
 		if ( apply_filters( 'jetpack_sync_listener_should_load', true ) ) {
 			self::initialize_listener();
 		}
-		
+
 		add_action( 'init', array( __CLASS__, 'add_sender_shutdown' ), 90 );
 
 	}
 
+	static function maybe_start_initial_sync() {
+		if ( ! self::can_do_initial_sync() ) {
+			return;
+		}
+		// Previous
+		$previous_version_and_time = Jetpack_Options::get_option( 'old_version', 0 );
+		$previous_version          = explode( ':', $previous_version_and_time );
+		self::do_initial_sync( JETPACK__VERSION, $previous_version[0], true );
+	}
+
+	static function can_do_initial_sync( $current_blog_id = null, $current_time = null ) {
+		if ( empty( $current_blog_id ) ) {
+			$current_blog_id = get_current_blog_id();
+		}
+		if ( empty( $current_time ) ) {
+			$current_time = time();
+		}
+
+		$version_with_time = explode( ':', Jetpack_Options::get_option( 'version', 0 ) );
+		if ( ! isset( $version_with_time[1] ) ) {
+			// This is not very likely to happen.
+			// lets set it to 0 so that the update happends right away
+			$version_with_time[1] = 0;
+		}
+		$version_updated = $version_with_time[1];
+
+		/**
+		 * Allows the dev to change the number of blogs that the nework is allowed update per second.
+		 * By default this value is set to 10 blogs per second.
+		 * The blogs blog_id determins if a site can update.
+		 *
+		 * @since 4.5.0
+		 *
+		 * @param int the number of blogs per second that should be allowed to update.
+		 */
+		$blogs_per_seconds = (int) apply_filters( 'jetpack_network_ramp_up_blogs_per_second', self::NETWORK_UPDATE_RAMP_UP_BLOGS_PER_SECOND );
+		$time_difference   = ( $current_time - $version_updated );
+
+		return ( $current_blog_id <= ( $time_difference * $blogs_per_seconds ) );
```

Comment:
I don't think many that there are not many network that run jetpack and also ran 100 million sites. I hope if they are running Jetpack that they also do their own testing like we do before deploying large core changes. 

That being said, you do make a point that there could be a delay in how fast site can trigger the autoupdate mechanism. For example a 100 000 sites network that runs Jetpack would take 27 hours. Which is long time. They could update the filter and it could easily reduce that number by increasing the number to 100 sites per second value. it would only take 3 hours. 

The reason why I like this solutions is that it doesn't require any sort of cron. It depends on the sites being used so sites that do not get any hits are ignored. There is minimal chance or running into race conditions. (Multi sites trying to write to the same value in the DB or cache) 

We could add some other filter that would allow devs to create their own function. If a site is allowed to for an update full sync or not.